components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -97,9 +97,86 @@ func InjectReconciliationLock(meta *metav1.ObjectMeta) error {
 	return nil
 }
 
+// resourceConfig holds parsed and validated OAuth proxy resource configuration
+type resourceConfig struct {
+	cpuRequest    resource.Quantity
+	memoryRequest resource.Quantity
+	cpuLimit      resource.Quantity
+	memoryLimit   resource.Quantity
+}
+
+// parseAndValidateAuthSidecarResources parses OAuth proxy resource annotations, applies defaults,
+// and validates that requests <= limits. This centralizes all resource parsing and validation logic.
+func parseAndValidateAuthSidecarResources(notebook *nbv1.Notebook) (*resourceConfig, error) {
+	// Start with defaults
+	config := &resourceConfig{
+		cpuRequest:    resource.MustParse(DefaultAuthSidecarCPURequest),
+		memoryRequest: resource.MustParse(DefaultAuthSidecarMemoryRequest),
+		cpuLimit:      resource.MustParse(DefaultAuthSidecarCPULimit),
+		memoryLimit:   resource.MustParse(DefaultAuthSidecarMemoryLimit),
+	}
+
+	annotations := notebook.GetAnnotations()
+	if annotations == nil {
+		return config, nil
+	}
+
+	// Parse and validate each annotation, updating config if valid
+	resourceMap := map[string]*resource.Quantity{
+		AnnotationAuthSidecarCPURequest:    &config.cpuRequest,
+		AnnotationAuthSidecarMemoryRequest: &config.memoryRequest,
+		AnnotationAuthSidecarCPULimit:      &config.cpuLimit,
+		AnnotationAuthSidecarMemoryLimit:   &config.memoryLimit,
+	}
+
+	for key, quantity := range resourceMap {
+		if valStr, ok := annotations[key]; ok && valStr != "" {
+			val, err := resource.ParseQuantity(strings.TrimSpace(valStr))
+			if err != nil {
+				return nil, fmt.Errorf("invalid value for annotation '%s': '%s': %w", key, valStr, err)
+			}
+			if val.Sign() < 0 {
+				return nil, fmt.Errorf("annotation '%s' value '%s' cannot be negative", key, valStr)
+			}
+			*quantity = val
+		}
+	}
+
+	// Validate that requests <= limits (including defaults)
+	if config.cpuRequest.Cmp(config.cpuLimit) > 0 {
+		return nil, fmt.Errorf("CPU request (%s) cannot be greater than CPU limit (%s)",
+			config.cpuRequest.String(), config.cpuLimit.String())
+	}
+
+	if config.memoryRequest.Cmp(config.memoryLimit) > 0 {
+		return nil, fmt.Errorf("memory request (%s) cannot be greater than memory limit (%s)",
+			config.memoryRequest.String(), config.memoryLimit.String())
+	}
+
```

Comment:
unable to find the changes suggested here https://github.com/opendatahub-io/kubeflow/pull/688#discussion_r2337490147
is this implemented somewhere else ?